src/main/java/io/vertx/ext/jdbc/impl/actions/JDBCStatementHelper.java

```diff
@@ -40,7 +59,7 @@
 
   private static final JsonArray EMPTY = new JsonArray(Collections.unmodifiableList(new ArrayList<>()));
 
-  private static final Pattern DATETIME = Pattern.compile("^\\d{4}-(?:0[0-9]|1[0-2])-[0-9]{2}T\\d{2}:\\d{2}:\\d{2}(\\.\\d{3})?Z$");
```

Comment:
The reason for this is that most JS engines date resolution is based on a 64bit int since the epoch so when encoding to json the nanos will be lost.